packages/cli-repl/src/cli-repl.ts

```diff
@@ -310,10 +311,14 @@ class CliRepl {
    * The greeting for the shell.
    */
   greet(): void {
-    const { version } = require('../package.json');
+    const { version, engines } = require('../package.json');
     console.log(`Using MongoDB: ${this.internalState.connectionInfo.buildInfo.version}`);
     console.log(`${clr('Using Mongosh Beta', ['bold', 'yellow'])}: ${version}`);
     console.log(`${MONGOSH_WIKI}`);
+    if (semver.lt(process.version, engines.node.replace('^', ''))) {
+      console.log(`WARNING: mismatched node version. Minimum node version required ${engines.node}. Currently using ${process.version}. Exiting...\n\n`);
```

Comment:
Maybe let's throw an `MongoshUnsupportedError` here?

Message is really good though -- super clear with node versions, hopefully will lead to less bug reports.

Reply:
We don't have unsupported, we only have unimplemented, which I think doesn't match this use case. I can change it to `RuntimeError`